git-delete-branch: add page (#6011)
tldr-pages/tldr: pages/common/git-delete-branch.md

# git delete-branch

> Delete local and remote Git branches.

> > Part of `git-extras`. You will get errors if you are checked out on a branch you are deleting.

Reviewer: ```suggestion
> Part of `git-extras`. An error occurs on deleting the current checked out branch.
```

Author: ```suggestion
> Part of `git-extras`. Errors will occur if you are deleting a checked out branch
```

Reviewer 2: I agree with the removal of "you", but it seems wrong to put on the same line as "Part of `git-extras`".  I've venture to say that this is note is unnecessary, because it's not likely that a user will think that they _can_ delete the current branch.  Thoughts?

Author: A user might make a mistake and not realise they are checked out on a particular branch

Reviewer 3: Well actually as I've just encountered with #5988, it does *not* give you an error if you try it...

Author: error: Cannot delete branch 'testremove' checked out

I have a test repo just for me and while checkedout, I ran this command and got this

> error: Cannot delete branch 'testremove' checked out

Author: it does however delete the remote branch since you are not _checked out_ at origin/testremove

Reviewer 3: Just saw that I got the message too. However it deletes all remote branches. I'd prefer not to include the note above, since it may let the user think it's save to run.

Reviewer 2: So should we resolve this thread?

Reviewer 3: I actually think we should remove the second sentense completely. We also don't include in every Git page, that errors will occur if you're not in a Git repository.

Author: ```suggestion
> Part of `git-extras`.
```

Author: But it is an easy error to make, many will make it

Reviewer 3: How about:
```suggestion
> Part of `git-extras`. When deleting the checked out branch, only the remote branch will be deleted.
```

Author: ```suggestion
> Part of `git-extras`. When deleting a checked out branch, only the remote branch will be deleted.
```

Reviewer 3: Can you have multiple checked out branches?

Author: Not to my knowledge, however the `the` seems out of place, a fits better and makes it read better

Reviewer 4: No, you can't have multiple checked out branches under normal operation to my knowledge.

@bl-ue: On the contrary, I think it's important to keep the `Part of git-extras.`, just in case someone confuses it for an inbuilt git command.

Reviewer 2: I meant that the note about deleting the checked out branch isn't necessary, @sbrl.

Author: Can we reach a result, what should suggestion should  I apply

> > Part of `git-extras`. When deleting a checked out branch, only the remote branch will be deleted.

Reviewer 2: There can only be one checked out branch, so I'd adjust the sentence like so:
```suggestion
> Part of `git-extras`. If deleting the checked out branch, only the remote branch will be deleted.
```

Author: I reallly dislike the usage of the `the`, it makes more nouny rather than the verb  status it deserves. Its start to explain :smile:

Reviewer 2: I'm not really sure what you mean; `a` and `the` are both articles and the only difference between them is the number of the nouns that they're used with.  Using "the" is nothing but more correct here.

Author: https://github.com/tldr-pages/tldr/pull/6011#discussion_r637851077 @marchersimon seemed to agree with me. Need a more solid opinion rather than +1

Reviewer 3: I also think `the` makes more sense here, since there's only *the* one checked out branch. However `a` is fine for me too and it's still your PR.

Author: Committedd

> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-delete-branch>.

- Delete a local and remote Git branch:

`git delete-branch {{branch_name}}`

- Delete multiple local and remote Git branches:

> `git delete-branch {{branch_name branch_name ...}}`

Reviewer 5: ```suggestion
`git delete-branch {{branch_name1 branch_name2 ...}}`
```
